overlord/snapstate/handlers.go

```diff
@@ -522,6 +527,11 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// Restore configuration for the new revision if available
+	if err = config.RestoreConfigSnapshotMaybe(st, snapsup.Name(), snapsup.Revision()); err != nil {
```

Comment:
@niemeyer Ah! Thanks for giving this example, I didn't know / forgot about refresh to specifc revision! I've added such case to spread test and indeed it fails without that "isRevert" condition in the code. Please take again at the spread test and see if the behavior is sensible.